### ISSUE TITLE
[Question] Javac not found in Java task

### ISSUE THREAD

**Issue by xukaisjz** [apache/dolphinscheduler#15722]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

erro logs:
 "tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/4_7.sh: line 4: /bin/javac: No such file or directory
	/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/4_7.sh: line 4: /bin/java: No such file or directory"

console logs:
```
[INFO] 2024-03-16 09:57:27.974 +0800 - ***********************************************************************************************
[INFO] 2024-03-16 09:57:27.974 +0800 - *********************************  Execute task instance  *************************************
[INFO] 2024-03-16 09:57:27.974 +0800 - ***********************************************************************************************
[INFO] 2024-03-16 09:57:27.974 +0800 - The current java source code will begin to replace the placeholder: public class Test {
    public static void main(String[] args) {
        System.out.println("=======");
        System.out.println("1111");
    }
}
[INFO] 2024-03-16 09:57:27.975 +0800 - tenantCode: root, task dir:/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7
[INFO] 2024-03-16 09:57:27.975 +0800 - the java source code:/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/Test.java, will be write to the file: public class Test {
    public static void main(String[] args) {
        System.out.println("=======");
        System.out.println("1111");
    }
}
[INFO] 2024-03-16 09:57:27.976 +0800 - Final Shell file is: 
[INFO] 2024-03-16 09:57:27.976 +0800 - ****************************** Script Content *****************************************************************
[INFO] 2024-03-16 09:57:27.976 +0800 - #!/bin/bash
BASEDIR=$(cd `dirname $0`; pwd)
cd $BASEDIR
${JAVA_HOME}/bin/javac -classpath .:/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7 /tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/Test.java;${JAVA_HOME}/bin/java -classpath .:/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7 Test  
[INFO] 2024-03-16 09:57:27.976 +0800 - ****************************** Script Content *****************************************************************
[INFO] 2024-03-16 09:57:27.976 +0800 - Executing shell command : sudo -u root -i /tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/4_7.sh
[INFO] 2024-03-16 09:57:28.051 +0800 - process start, process id is: 1921
[INFO] 2024-03-16 09:57:29.051 +0800 -  -> 
	/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/4_7.sh: line 4: /bin/javac: No such file or directory
	/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7/4_7.sh: line 4: /bin/java: No such file or directory
[INFO] 2024-03-16 09:57:29.053 +0800 - process has exited. execute path:/tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7, processId:1921 ,exitStatusCode:127 ,processWaitForStatus:true ,processExitValue:127
[INFO] 2024-03-16 09:57:29.053 +0800 - java task run result: TaskResponse(varPool=null, processId=1921, resultString=null, appIds=null, process=null, cancel=false, exitStatusCode=127, status=null)
[INFO] 2024-03-16 09:57:29.053 +0800 - ***********************************************************************************************
[INFO] 2024-03-16 09:57:29.053 +0800 - *********************************  Finalize task instance  ************************************
[INFO] 2024-03-16 09:57:29.054 +0800 - ***********************************************************************************************
[INFO] 2024-03-16 09:57:29.055 +0800 - Upload output files: [] successfully
[INFO] 2024-03-16 09:57:29.059 +0800 - Send task execute status: FAILURE to master : 172.17.0.4:1234
[INFO] 2024-03-16 09:57:29.060 +0800 - Remove the current task execute context from worker cache
[INFO] 2024-03-16 09:57:29.060 +0800 - The current execute mode isn't develop mode, will clear the task execute file: /tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7
[INFO] 2024-03-16 09:57:29.060 +0800 - Success clear the task execute file: /tmp/dolphinscheduler/exec/process/root/12922454007776/12936536535008_4/4/7
[INFO] 2024-03-16 09:57:29.060 +0800 - FINALIZE_SESSION
```

### What you expected to happen

normal

### How to reproduce

![image](https://github.com/apache/dolphinscheduler/assets/68056796/873b2aca-5e69-4c6e-92e9-dd791a2fcb7c)

### Version

3.2.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by ruanwenjun** [comment]
Have you config `${JAVA_HOME}`?

**Comment by xukaisjz** [comment]
容器启动的，在容器中echo ${JAVA_HOME}是有值的

**Comment by ruanwenjun** [comment]
> 容器启动的，在容器中echo ${JAVA_HOME}是有值的
> […](#)

请检查 `${JAVA_HOME}/bin/javac` 路径是否正确

**Comment by Gallardot** [comment]
The [original image of dolphinscheduler worker uses jre](https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-worker/src/main/docker/Dockerfile#L18C27-L18C27), which does not include javac by default. Here, jdk should be used. I'll try to fix this issue.

**Comment by Gallardot** [comment]
@xukaisjz I have fixed this issue in #15764 . You can have a try.
